src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,78 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  buttonLoader: boolean;
+  addFunction: (newComment: CommentData) => Promise<void>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  buttonLoader,
+  addFunction = () => {},
+}) => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+    setNameError(false);
+  };
+
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(event.target.value);
+    setEmailError(false);
+  };
+
+  const [text, setText] = useState('');
+  const [textError, setTextError] = useState(false);
+
+  const handleTextChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setText(event.target.value);
+    setTextError(false);
+  };
+
+  const normalizeValue = (value: string) => {
+    return value.trim();
+  };
+
+  const onSubmit = () => {
```

Comment:
The `onSubmit` function is missing the `event` parameter. You need to include `event` as a parameter to use `event.preventDefault()` correctly.